### PR TITLE
refactor(wifi_scan)!: not having separate "can" APIs for checks

### PR DESCRIPTION
**Pull request by daadu**
- instead `Result<ValueType, ErrorType>` is added and used as return type
 - example app simplified
 - Kotlin upgrade

### CODE REVIEW

File: packages/wifi_scan/lib/wifi_scan.dart [hunk 1]
```diff
@@ -18,73 +18,66 @@ class WiFiScan {
...
-  /// Checks if it is ok to invoke [startScan].
-  ///
-  /// Necesearry platform requirements, like permissions dependent services,
-  /// configuration, etc are checked.
+  /// Request a Wi-Fi scan.
   ///
-  /// Set [askPermissions] flag to ask user for necessary permissions.
-  Future<CanStartScan> canStartScan({bool askPermissions = true}) async {
-    final canCode = await _channel.invokeMethod<int>("canStartScan", {
+  /// Returns null if successful, else [StartScanErrors].
```
Comment (Hallot) [inline packages/wifi_scan/lib/wifi_scan.dart:26]: I am more use to have the "OK" value part of the enum, but then C does not have null types, so...  
Both styles are fine to me, just not sure whether there exists some consensus on this.

**Comment by Hallot** [comment]
```
E/MethodChannel#wifi_scan( 6508): Failed to handle method call
E/MethodChannel#wifi_scan( 6508): java.util.NoSuchElementException: Key error is missing in the map.
E/MethodChannel#wifi_scan( 6508): 	at kotlin.collections.MapsKt__MapWithDefaultKt.getOrImplicitDefaultNullable(MapWithDefault.kt:24)
E/MethodChannel#wifi_scan( 6508): 	at kotlin.collections.MapsKt__MapsKt.getValue(Maps.kt:346)
E/MethodChannel#wifi_scan( 6508): 	at dev.flutternetwork.wifi.wifi_scan.WifiScanPlugin.onMethodCall(WifiScanPlugin.kt:189)
E/MethodChannel#wifi_scan( 6508): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
E/MethodChannel#wifi_scan( 6508): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:178)
E/MethodChannel#wifi_scan( 6508): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$handleMessageFromDart$0$DartMessenger(DartMessenger.java:206)
E/MethodChannel#wifi_scan( 6508): 	at io.flutter.embedding.engine.dart.-$$Lambda$DartMessenger$6ZD1MYkhaLxyPjtoFDxe45u43DI.run(Unknown Source:12)
E/MethodChannel#wifi_scan( 6508): 	at android.os.Handler.handleCallback(Handler.java:938)
E/MethodChannel#wifi_scan( 6508): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/MethodChannel#wifi_scan( 6508): 	at android.os.Looper.loopOnce(Looper.java:201)
E/MethodChannel#wifi_scan( 6508): 	at android.os.Looper.loop(Looper.java:288)
E/MethodChannel#wifi_scan( 6508): 	at android.app.ActivityThread.main(ActivityThread.java:7839)
E/MethodChannel#wifi_scan( 6508): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#wifi_scan( 6508): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/MethodChannel#wifi_scan( 6508): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/flutter ( 6508): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: PlatformException(error, Key error is missing in the map., null, java.util.NoSuchElementException: Key error is missing in the map.
E/flutter ( 6508): 	at kotlin.collections.MapsKt__MapWithDefaultKt.getOrImplicitDefaultNullable(MapWithDefault.kt:24)
E/flutter ( 6508): 	at kotlin.collections.MapsKt__MapsKt.getValue(Maps.kt:346)
E/flutter ( 6508): 	at dev.flutternetwork.wifi.wifi_scan.WifiScanPlugin.onMethodCall(WifiScanPlugin.kt:189)
E/flutter ( 6508): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
E/flutter ( 6508): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:178)
E/flutter ( 6508): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$handleMessageFromDart$0$DartMessenger(DartMessenger.java:206)
E/flutter ( 6508): 	at io.flutter.embedding.engine.dart.-$$Lambda$DartMessenger$6ZD1MYkhaLxyPjtoFDxe45u43DI.run(Unknown Source:12)
E/flutter ( 6508): 	at android.os.Handler.handleCallback(Handler.java:938)
E/flutter ( 6508): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/flutter ( 6508): 	at android.os.Looper.loopOnce(Looper.java:201)
E/flutter ( 6508): 	at android.os.Looper.loop(Looper.java:288)
E/flutter ( 6508): 	at android.app.ActivityThread.main(ActivityThread.java:7839)
E/flutter ( 6508): 	at java.lang.reflect.Method.invoke(Native Method)
E/flutter ( 6508): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/flutter ( 6508): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
E/flutter ( 6508): )
E/flutter ( 6508): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:607:7)
E/flutter ( 6508): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:167:18)
E/flutter ( 6508): <asynchronous suspension>
E/flutter ( 6508): #2      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:367:43)
E/flutter ( 6508): <asynchronous suspension>
E/flutter ( 6508): #3      WiFiScan.getScannedResults (package:wifi_scan/wifi_scan.dart:45:23)
E/flutter ( 6508): <asynchronous suspension>
E/flutter ( 6508): #4      _MyAppState.build.<anonymous closure>.<anonymous closure> (package:wifi_scan_example/main.dart:135:36)
E/flutter ( 6508): <asynchronous suspension>
```

When doing "Scan" then "get".

File: packages/wifi_scan/lib/wifi_scan.dart (hunk 1 above)
Comment (daadu) [reply packages/wifi_scan/lib/wifi_scan.dart:26]: You mean having `StartScanErrors.none`, if no error instead of `null`?

Comment (Hallot) [reply packages/wifi_scan/lib/wifi_scan.dart:26]: Yes. But this is fine as well.

**Comment by daadu** [comment]
@Hallot resolved both issue with https://github.com/flutternetwork/WiFiFlutter/pull/246/commits/7a643ef772aadb6b874492da3c38b33d57168580 and https://github.com/flutternetwork/WiFiFlutter/pull/246/commits/d18139a9a94b4d07578618dfb7a98ba07b032fe4.

The `android:exported="true"` is added by [default by flutter tool when creating new project after v2.8.0](https://github.com/flutter/flutter/pull/91934/files) - therefore does not need any mention in README.

File: packages/wifi_scan/lib/wifi_scan.dart (hunk 1 above)
Comment (daadu) [reply packages/wifi_scan/lib/wifi_scan.dart:26]: with that, it is hard to incorporate something like `hasError` check with generic `Result<ValueType, ResultType>` class.

**Pull request merged**